tests/Unit/NumericalAlgorithms/LinearAlgebra/Test_FindGeneralizedEigenvalues.cpp

```diff
@@ -0,0 +1,170 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Matrix.hpp"
+#include "ErrorHandling/Error.hpp"
+#include "NumericalAlgorithms/LinearAlgebra/FindGeneralizedEigenvalues.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace {
+void test_find_generalized_eigenvalues(
+    const Matrix& A, const Matrix& B, const size_t eigenvector_size,
+    const size_t eigenvalue_size, const bool force_lapack_failure = false) {
+  // Set up vectors and a matrix to store the eigenvalues and eigenvectors
+  DataVector eigenvalues_real_part(eigenvalue_size, 0.0);
+  DataVector eigenvalues_im_part(eigenvalue_size, 0.0);
+  Matrix eigenvectors(eigenvector_size, eigenvector_size, 0.0);
+
+  // Get the generalized eigenvectors and eigenvalues
+  find_generalized_eigenvalues(&eigenvalues_real_part, &eigenvalues_im_part,
+                               &eigenvectors, A, B, force_lapack_failure);
+
+  // Expected eigenvalues and eigenvectors
+  // Eigenvectors normalized so last component is 1
+  constexpr double expected_larger_eigenvalue = 10.099019513592784;
+  constexpr double expected_smaller_eigenvalue = -0.09901951359278449;
+  constexpr double expected_larger_eigenvector_value = 0.819803902718557;
```

Comment:
It would be useful to have a comment why the `expected` eigenvectors are `double` not `DataVector`

Reply:
Please explain clearly that you are checking ratios of components of the eigenvectors